ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -332,16 +341,20 @@ public CancelHandle receiveSessionOpenRequest(@NotNull final SessionOpenRequest
                 return;
             }
 
-            Affinity clusterAffinity = getClusterAffinity();
-            if (clusterAffinity != null) {
-                sessionOpenRequest.updateStrongAffinity(clusterAffinity);
+            // do not update strongAffinity when it is of type NodeAffinity; this will be achieved on the client in DiscoveryInterceptor via targetAffinity
+            Affinity strongAffinity = getStrongAffinity(statefulSessionComponent);
+            if (strongAffinity != null && !(strongAffinity instanceof NodeAffinity)) {
+                sessionOpenRequest.updateStrongAffinity(strongAffinity);
```

Comment:
I should think we only need to invoke updateStrongAffinity(...) if the strong affinity changed, regardless of what type it is.

Reply:
Sorry I didn't see this earlier. Making changes to this at this stage is risky, even if it looks obvious. I'd rather leave what works and update in a separate PR.